Kraft v0.7.0 fails on Apple Silicon (Sonoma)

### Describe the bug

```
$ kraft fetch --log-type=basic
 W  specification in Kraftfile (v0.5) version is not latest (v0.6)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x105182354]

goroutine 1 [running]:
kraftkit.sh/internal/cli/kraft/fetch.(*FetchOptions).pull(0x14000415c00, {0x1058affb8, 0x14000287350}, {0x1058c6058, 0x1400031bb00}, {0x16bc4f31d, 0x8}, 0x1, 0xffffffffffffffff)
        /__w/kraftkit/kraftkit/internal/cli/kraft/fetch/fetch.go:216 +0x674
kraftkit.sh/internal/cli/kraft/fetch.(*FetchOptions).Run(0x14000415c00, {0x1058affb8, 0x14000287350}, {0x0?, 0x0?, 0x0?})
        /__w/kraftkit/kraftkit/internal/cli/kraft/fetch/fetch.go:384 +0x17c
kraftkit.sh/cmdfactory.New.func1(0x0?, {0x140001a92a0?, 0x0?, 0x0?})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:365 +0x4c
kraftkit.sh/cmdfactory.AttributeFlags.bind.func3(0x14000235a00?, {0x140001a92a0, 0x1, 0x2})
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:547 +0x150
github.com/spf13/cobra.(*Command).execute(0x14000160c00, {0x140001a9280, 0x2, 0x2})
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x658
github.com/spf13/cobra.(*Command).ExecuteC(0x14000160000)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:985
kraftkit.sh/cmdfactory.Main({0x1058affb8?, 0x140003a2f60}, 0x14000160000)
        /__w/kraftkit/kraftkit/cmdfactory/builder.go:183 +0x70
kraftkit.sh/internal/cli/kraft.Main({0x16bc4f300?, 0x1041eb6e0?, 0x10647d228?})
        /__w/kraftkit/kraftkit/internal/cli/kraft/kraft.go:165 +0x51c
main.main()
        /__w/kraftkit/kraftkit/cmd/kraft/main.go:19 +0x94
make: *** [unikraft] Error 2
```

### Steps to reproduce

Run `Kraft fetch`

### Expected behavior

_No response_

### Which architectures were you using or does this bug affect?

x86_64

### Which operating system were you using or does this bug affect?

macOS

### Relevant log output

_No response_

--- a/internal/cli/kraft/events/events.go
+++ b/internal/cli/kraft/events/events.go
@@ -72,6 +72,8 @@ func (opts *EventOptions) Pre(cmd *cobra.Command, _ []string) error {
 func (opts *EventOptions) Run(ctx context.Context, args []string) error {
 	var err error
 
+	log.G(ctx).Warnf("This command is DEPRECATED and should not be used")
+
 	ctx, cancel := context.WithCancel(ctx)
 	platform := mplatform.PlatformUnknown
 

--- a/internal/cli/kraft/fetch/fetch.go
+++ b/internal/cli/kraft/fetch/fetch.go
@@ -355,6 +355,8 @@ func (opts *FetchOptions) pull(ctx context.Context, project app.Application, wor
 }
 
 func (opts *FetchOptions) Run(ctx context.Context, _ []string) error {
+	log.G(ctx).Warnf("This command is DEPRECATED and should not be used")
+
 	// Filter project targets by any provided CLI options
 	selected := opts.project.Targets()
 

--- a/internal/cli/kraft/set/set.go
+++ b/internal/cli/kraft/set/set.go
@@ -41,6 +41,7 @@ import (
 	"github.com/spf13/cobra"
 
 	"kraftkit.sh/cmdfactory"
+	"kraftkit.sh/log"
 	"kraftkit.sh/packmanager"
 	"kraftkit.sh/unikraft/app"
 )
@@ -97,6 +98,8 @@ func (*SetOptions) Pre(cmd *cobra.Command, _ []string) error {
 
 func (opts *SetOptions) Run(ctx context.Context, args []string) error {
 	var err error
+
+	log.G(ctx).Warnf("This command is DEPRECATED and should not be used")
 
 	workdir := ""
 	confOpts := []string{}

--- a/internal/cli/kraft/unset/unset.go
+++ b/internal/cli/kraft/unset/unset.go
@@ -40,6 +40,7 @@ import (
 	"github.com/spf13/cobra"
 
 	"kraftkit.sh/cmdfactory"
+	"kraftkit.sh/log"
 	"kraftkit.sh/packmanager"
 	"kraftkit.sh/unikraft/app"
 )
@@ -95,6 +96,8 @@ func (*UnsetOptions) Pre(cmd *cobra.Command, _ []string) error {
 
 func (opts *UnsetOptions) Run(ctx context.Context, args []string) error {
 	var err error
+
+	log.G(ctx).Warnf("This command is DEPRECATED and should not be used")
 
 	workdir := ""
 	confOpts := []string{}

